src/parquet/column_reader.cc

```diff
@@ -212,6 +121,9 @@ bool ColumnReader::ReadNewPage() {
       // Read a data page.
       num_buffered_values_ = current_page_header_.data_page_header.num_values;
```

Comment:
I missed that part of the spec. How annoying! 

Perhaps the API should be then:

`size_t ReadBatch(int batchsize, int16_t* def_levels, int16_t* rep_levels, T* values, size_t* values_read)`

Since you will have to inspect the definition levels (using the maximum definition level -- needs to be passed in) to determine how many encoded values are available. thoughts?